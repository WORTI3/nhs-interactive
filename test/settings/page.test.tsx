import Page from "../../app/settings/page";
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { load } from "cheerio";

describe('Settings', () => {
  test('rendered component should match snapshot', () => {
    const { container } = render(<Page />);
    expect(container).toMatchSnapshot();
  });

  test('should render the component with the correct links & elements', () => {
    const { container, getByText } = render(<Page />);
    const html = container.innerHTML;
    const $ = load(html);
  
    expect($('h1').length).toBe(1);
    expect($('h1').text()).toBe('Settings BETA');
    expect($('h2').length).toBe(1);
    expect($('h2').text()).toBe('Here you can update the OpenAI API Key');
    expect($('a').length).toBe(3);
    expect($('svg').length).toBe(1);
    expect($('p').length).toBe(2);
    expect(getByText('Show Settings')).toBeInTheDocument();
  });
});