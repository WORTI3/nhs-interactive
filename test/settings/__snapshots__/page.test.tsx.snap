// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings rendered component should match snapshot 1`] = `
<div>
  <div
    class="nhsuk-back-link"
  >
    <a
      class="nhsuk-back-link__link"
      href="/"
    >
      <svg
        aria-hidden="true"
        class="nhsuk-icon nhsuk-icon__chevron-left"
        height="32"
        viewBox="0 0 24 24"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.5 12c0-.3.1-.5.3-.7l5-5c.4-.4 1-.4 1.4 0s.4 1 0 1.4L10.9 12l4.3 4.3c.4.4.4 1 0 1.4s-1 .4-1.4 0l-5-5c-.2-.2-.3-.4-.3-.7z"
        />
      </svg>
      Go back
    </a>
  </div>
  <h1>
    Settings 
    <strong
      class="nhsuk-tag nhsuk-tag--white align-middle"
    >
      <span
        class="text-transparent bg-clip-text bg-gradient-to-r from-blue-700 via-blue-500 to-purple-700"
      >
        BETA
      </span>
    </strong>
  </h1>
  <h2>
    Here you can update the OpenAI API Key
  </h2>
  <p>
    You can obtain your personal API key 
    <a>
      here
    </a>
    .
  </p>
  <div
    class="nhsuk-inset-text"
  >
    <span
      class="nhsuk-u-visually-hidden"
    >
      Information: 
    </span>
    <p>
      To change back to your default configuration set in your 
      <span
        class="font-semibold"
      >
        .env
      </span>
       file. Use the API key: 
      <span
        class="font-semibold"
      >
        local
      </span>
      .
    </p>
  </div>
  <a
    aria-disabled="false"
    class="nhsuk-button"
    draggable="false"
    role="button"
  >
    Show
     Settings
  </a>
</div>
`;
